src/bp/core/services/middleware/state-manager.ts

```diff
@@ -122,14 +123,22 @@ export class StateManager {
   }
 
   public async persist(event: sdk.IO.IncomingEvent, ignoreContext: boolean) {
-    const { workflows } = event.state.session
+    const { workflows, lastMessages } = event.state.session
 
     for (const wf of Object.keys(workflows)) {
       workflows[wf].variables = _.mapValues(workflows[wf].variables, (x: sdk.BoxedVariable<any>) => x.unbox()) as any
     }
 
     const sessionId = SessionIdFactory.createIdFromEvent(event)
 
+    if (!lastMessages.find(x => x.eventId === event.id)) {
+      addLogToEvent(`No message was sent by the bot`, event)
```

Comment:
Sorry this is my OCD

No message was ==> No messages were

In english 0 uses plural form